src/_pytest/runner.py

```diff
@@ -216,6 +217,9 @@ def __init__(self, func, when, treat_keyboard_interrupt_as_exception=False):
             else:
                 self.stop = time()
                 raise
+        except Exit:
```

Comment:
this change makes the api boundaries of CallInfo even more of a problem
i would like to propose reshaping callinfo a bit before adding that to it

Reply:
Well, I assume that might take some time?

This is an rather old issue (talked about in 2016 already).
Do you have a good suggestion for https://github.com/pytest-dev/pytest/pull/4280 then?

Reply:
i got a pr for the change almost ready, will push this evening

Reply:
For referrence, I am assuming it is this one: https://github.com/pytest-dev/pytest/pull/4299

Reply:
it is - i still have no idea why it is broken now ^^ its bound to be very subtle